src/battle-animations.ts

```diff
@@ -948,6 +970,13 @@ class BattleScene {
 		if (!this.animating) return;
 		let isIntense = false;
 		let weather = this.battle.weather;
+		for (const side of this.battle.sides) {
+			for (const active of side.active) {
+				if (active && !active.fainted && ['Air Lock', 'Cloud Nine'].includes(active.ability)) {
```

Comment:
Considering all the places we need to check for Neutralizing Gas and Gastro Acid, we probably want a `pokemon.effectiveAbility()` function.

Reply:
Which can also check `.fainted`, saving us another thing here.